hcl2template/types.packer_config.go

```diff
@@ -409,40 +361,40 @@ func (cfg *PackerConfig) recursivelyEvaluateDatasources(ref DatasourceRef, depen
 				"sources. Either your data source depends on more than ten " +
 				"other data sources, or your data sources have a cyclic " +
 				"dependency. Please simplify your config to continue. ",
+			Subject: &ds.block.DefRange,
```

Comment:
Since we are returning early I would recommend building the Subject in the conditional block and move line 385 right about 387. This way we don't have to track back to0 far to see where `ds` is declared and initialized to.